action_server/src/robocorp/action_server/_server_expose.py

```diff
@@ -95,6 +95,30 @@ def forward_request(base_url: str, payload: BodyPayload) -> requests.Response:
         raise NotImplementedError(f"Method {payload.method} not implemented")
 
 
+async def handle_ping_pong(
+    ws: websockets.WebSocketClientProtocol,
+    pong_queue: asyncio.Queue,
+    no_connection_queue: asyncio.Queue,
+):
+    while True:
+        await asyncio.sleep(2)
+        log.debug("️➡️ outgoing ping")
+        await ws.send("ping")
```

Comment:
This might fail if the connection is broken - @ovatto points out.

Reply:
Then we will catch it outside the websocket connection context and retry.

Reply:
Also, if the connection is broken the `websocket.connect` will throw, so this won't happen.